README.md

```diff
@@ -0,0 +1,188 @@
+This demo shows how to migrate a Spring Boot application to Quarkus, and how to measure the performance gains along the way. It can all be live-coded, showing how easy migrating can be.
+
+## Preparation
+
+The supporting scripts for this demo assume MacOS. You will need the `http` utility:
+
+```shell
+brew install httpie
+```
+
+You will also need a container runtime, such as Podman or Docker.
+
+Finally, you will need `jq`. 
+
+## The Spring Boot Application 
+
+
+Start with `quarkus-spring-todo-app`. This is the [Quarkus todo sample](https://github.com/cescoffier/quarkus-todo-app), rewritten to Spring Boot 3.3.0. There is a single test implemented using Test Containers and Rest Assured (using Rest Assured isn't strictly idiomatic Spring, but it makes things easier later on.) For examples of migrating tests, see [Eric Deandrea](https://github.com/edeandrea)'s comprehensive "Spring to Quarkus" [code samples](https://github.com/quarkus-for-spring-developers/examples). 
+
+### JVM mode
+
+Package it and launch it:
+
+```
+cd spring-todo-app
+./mvnw package
+```
+
+Before going further, start the database that the live instance will be using. Once that's done, start the app.
+
+```
+just start-infra
+java -jar target/spring-todo-app-0.0.1-SNAPSHOT.jar -Xmx600m -Xms600m
+```
+Setting the fixed `-Xmx` and `-Xms` is important to reduce warmup effects and give apples-to-apples comparisons.
+
+Now it's time for a mini-stress test, by running a [defined set of actions](https://github.com/cescoffier/spring-to-quarkus-demo/blob/main/justfile#L15). Before every run, make sure you have an empty database, by running `just restart-infra`. In a new window, at the top-level of the repository, run
+
+```
+just stress
+```
+
+Collect the startup time and footprint after the stress actions complete. 
+You can read the startup time from the application log. It will be something like 
+
+```
+2024-06-27T11:56:59.475+01:00  INFO 37043 --- [           main] m.escoffier.spring.todo.TodoApplication  : Started TodoApplication in 2.779 seconds (process running for 3.247)
+```
+
+When measuring the footprint of a Java application, you should measure Resident Set Size (RSS) and not the JVM heap size which is only a small part of the overall problem. The JVM not only allocates native memory for heap (-Xms, -Xmx) but also structures required by the jvm to run your application, such as compiled code and class metadata.
+
+On MacOS, to measure the RSS, run
+
+```shell
+ps aux | grep -i java | grep -i todo | grep -v grep | awk {'print $2'} | xargs ps x -o pid,rss,command -p
```

Comment:
pgrep $1 | xargs ps -o pid,rss,command -p | awk '{$2=int($2/1024)"M";}{ print;}' because Mb is easier to read?

Reply:
I couldn't get the `pgrep` to work from a different terminal window, but I've added the nice MB awk conversion, thanks.

```
ps aux | grep -i java | grep -i todo | grep -v grep | awk {'print $2'} | xargs ps x -o pid,rss,command -p | awk '{$2=int($2/1024)"M";}{ print;}
```